src/Mealz/MealBundle/Tests/AbstractDatabaseTestCase.php

```diff
@@ -127,61 +123,38 @@ protected function createProfile()
         return $profile;
     }
 
-    protected function createLogin(Profile $profile = null)
-    {
-        $name = $profile ? $profile->getName() : rand();
-        $login = new Login();
-        $login->setUsername($name);
-        $login->setSalt(md5(uniqid(null, true)));
-
-        /** @var Symfony\Component\Security\Core\Encoder\PasswordEncoderInterface $encoder */
-        $encoder = static::$kernel->getContainer()->get('security.encoder_factory')->getEncoder($login);
-        $login->setPassword($encoder->encodePassword($name, $login->getSalt()));
-
-        if ($profile) {
-            $login->setProfile($profile);
-        }
-
-        return $login;
-    }
-
-    /**
-     * @return Category
-     */
-    protected function createCategory()
+    protected function createCategory(): Category
     {
+        $uniqSuffix = mt_rand();
         $category = new Category();
-        $category->setTitleDe('Title DE '.rand());
-        $category->setTitleEn('Title EN '.rand());
+        $category->setTitleDe('Title DE '.$uniqSuffix);
+        $category->setTitleEn('Title EN '.$uniqSuffix);
 
         return $category;
     }
 
-    /**
-     * @param array $entities
-     */
-    public function persistAndFlushAll($entities)
+    public function persistAndFlushAll(array $entities): void
     {
+        /** @var EntityManagerInterface $entityManager */
         $entityManager = $this->getDoctrine()->getManager();
 
         $entityManager->transactional(
-            function ($entityManager) use ($entities) {
-                /** @var EntityManager $entityManager */
+            static function (EntityManagerInterface $entityManager) use ($entities) {
                 // transaction is need for Participant entities
                 foreach ($entities as $entity) {
                     $entityManager->persist($entity);
                 }
+
                 $entityManager->flush();
             }
         );
     }
 
-    /**
-     * @param Loader $loader
-     */
-    protected function push($loader)
+    protected function push(Loader $loader): void
     {
-        $entityManager = static::$kernel->getContainer()->get('doctrine')->getManager();
+        /** @var EntityManagerInterface $entityManager */
```

Comment:
@var kann an doch ersetzen durch EntityManagerInterface $entityManager in der Zeile drunter, richtig?

Reply:
Meinst du so?

```EntityManagerInterface $entityManager = $this->getDoctrine()->getManager();```

Es geht in PHP nicht. Oder meinst du etwas anderes?

Reply:
Ach ja, stimmt. Sorry! getManager() gibt ja einen ObjectManager zurück.